test/test-framework-spec.md

```diff
@@ -0,0 +1,857 @@
+# Ballerina Test Framework Specification
+
+**Contributors** :
+Joseph Fonseka, Asma Jabir, Aquib Zulfikar, Fathima Dilhasha
```

Comment:
@hevayo Shall we add initial contributions as well?

Reply:
Yeh +1

Reply:
Removed the contirbutors to be consistent with other spec documents